File: tests/fixtures/route_extraction_smoke/service-a/src/main/java/smoke/a/UserController.java
package smoke.a;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class UserController {
    /** Same literal path as smoke.b.UserController — Route ids must still differ by microservice. */
    @GetMapping("/api/users")
    public String users() {
        return "";
    }
}

File: tests/fixtures/mixed-jvm/src/main/kotlin/com/foo/UserController.kt
package com.foo

import org.springframework.web.bind.annotation.RestController
import org.springframework.web.bind.annotation.GetMapping

/**
 * Kotlin controller that constructor-injects the Java [UserService] and calls it.
 * Spring ``@RestController`` must map to role CONTROLLER (detector reused on Kotlin).
 */
@RestController
class UserController(val userService: UserService) {

    @GetMapping("/users")
    fun get(): String {
        return userService.getById(1L)
    }
}
